src/components/Swap/RoutingDiagram/utils.ts

```diff
@@ -30,7 +27,7 @@ export function getTokenPath(trade: InterfaceTrade<Currency, Currency, TradeType
       const entry: RoutingDiagramEntry['path'][0] = [
         tokenIn,
         tokenOut,
-        nextPool instanceof Pair ? V2_DEFAULT_FEE_TIER : nextPool.fee,
+        nextPool instanceof Pool ? nextPool.fee : /* V2 FeeAmount= */ FeeAmount.MEDIUM,
```

Comment:
this logic is repeated in two diff places (as shown below) and requires some specific knowledge imo. maybe worth factoring out somehow